src/Parquet/Utils.hs

```diff
@@ -17,7 +17,7 @@ failOnMay ::
   ) =>
   Maybe a -> String -> m a
 failOnMay Nothing s = fail s
```

Comment:
Looks like the `failOnMay` and `failOnExcept` functions defined here are only used once throughout the project, could we have them inlined where they're used instead? Or maybe we could define them as local functions, up to you.

Reply:
I plan to use them in a couple of upcoming commits (already in progress). I'll inline them as per your suggestion in case I end up not using them.